### PR TITLE
Common Backup Codes UI component in My Account

### PR DESCRIPTION
**Pull request by chaminjay**
#### Preview
image

### Related Issues
- Support obtaining backup codes for SMS OTP and Email OTP via the My Account `#188`

### Related PRs
- n/a

### Checklist
- [x] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [x] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

**Review by pavinduLakshan — DISMISSED** [review]

**Review by JayaShakthi97 — DISMISSED** [review]

File: apps/myaccount/src/components/multi-factor-authentication/multi-factor-authentication.tsx [hunk 1]
```diff
@@ -56,10 +57,11 @@ export const MultiFactorAuthentication: React.FunctionComponent<MfaProps> = (pro
     } = props;
 
     const { t } = useTranslation();
-    const dispatch = useDispatch();
+    const dispatch: Dispatch<any> = useDispatch();
```
Comment (brionmario) [inline apps/myaccount/src/components/multi-factor-authentication/multi-factor-authentication.tsx:60]: No need to have any here. `Dispatch` internally adds `AnyAction`.

```suggestion
    const dispatch: Dispatch = useDispatch();
```

File: apps/myaccount/src/configs/ui.ts [hunk 2]
```diff
@@ -209,62 +212,62 @@ export const getEmptyPlaceholderIllustrations = (): {
...
         error: ErrorIcon,
         info: InfoIcon,
         success: SuccessIcon,
         warning: WarningIcon
     };
 };
- 
-export const AppSwitcherIcons = (): Record<string, any> => { 
+
+export const AppSwitcherIcons = (): Record<string, any> => {
```
Comment (brionmario) [inline apps/myaccount/src/configs/ui.ts:260]: Shall we try this type for better IntelliSense.
```suggestion
export const AppSwitcherIcons = (): {
    console: React.FunctionComponent<React.SVGProps<SVGSVGElement>>;
    myAccount: React.FunctionComponent<React.SVGProps<SVGSVGElement>>;
 } => {
```

**Review by brionmario — APPROVED** [review]

File: apps/myaccount/src/components/multi-factor-authentication/multi-factor-authentication.tsx [hunk 3]
```diff
@@ -210,20 +212,38 @@ export const MultiFactorAuthentication: React.FunctionComponent<MfaProps> = (pro
...
-                                        onBackupFlowCompleted={ handleBackupCodeFlowCompleted }
-                                        handleSessionTerminationModalVisibility={
-                                            () => setShowSessionTerminationModal(true)
-                                        }
-                                    />
-                                ) : null
-                            }
                         </List.Item>
                     ) : null }
+                { hasRequiredScopes(featureConfig?.security, featureConfig?.security?.scopes?.read, allowedScopes) &&
```
Comment (brionmario) [inline apps/myaccount/src/components/multi-factor-authentication/multi-factor-authentication.tsx:217]: This feature should be able to be disabled by passing a key in to the `disabled_features` array of the feature config. Make sure to test that as well.

Ref: https://github.com/wso2/identity-apps/blob/9cac913a65c27205ed05807d05ba6e10047f4010/apps/myaccount/src/components/multi-factor-authentication/multi-factor-authentication.tsx#L160

**Review by brionmario — DISMISSED** [review]

File: apps/myaccount/src/components/multi-factor-authentication/multi-factor-authentication.tsx (hunk 3 above)
Comment (chaminjay) [reply apps/myaccount/src/components/multi-factor-authentication/multi-factor-authentication.tsx:217]: ack

**Review by brionmario — DISMISSED** [review]

**Review by pavinduLakshan — DISMISSED** [review]

**Review by janakamarasena — APPROVED** [review]

**Pull request merged**